text/0000-add-cow-to-prelude.md

```diff
@@ -0,0 +1,138 @@
+refers to in the prelude. If a crate defines its own type called `Ok`, it will
+likely be confusing to readers, and will break any macros which don't refer to
+`std::result::Result::Ok` through its full path.
+
+Unfortunately, `Cow` is an extremely overloaded term in Rust today, so we can't
+reasonably export that identifier in the prelude. Applications today might be
+using this name to refer to:
+
+- California, Oregon, Washington (the most common usage)
+- Can of Worms
+- City of Waterloo
+- Crush of the Week
+- Container on Wheels
+- Cost of Waste
+
+to name a few. For this reason, the re-export of this type must be something
+which is unlikely or impossible to appear in Rust programs today. For this
+reason, the export will be defined as `pub use core::borrow::Cow as 🐄;`. As the
+`non_ascii_idents` feature is not yet stable, this alias cannot possibly shadow
+any type defined in stable Rust today. By defining this non-ascii identifier to
+mean `Cow` today, we can ensure that it is associated with this type, and does
+not become a common term for other uses as `Cow` clearly already is today.
+
+Every enum exported in the prelude today also exports its variants at the top
+level (e.g. `Option`, `Result`, etc). Clearly we should do the same for `🐄`.
+However, similar to how we can't reasonably begin exporting `Cow` from the
+prelude today, we clearly can't export `Borrowed` or `Owned` either (the
+alternative uses of these terms should be clear and thus are not listed here).
+
+This RFC proposes two potential alternatives for `Borrowed` and `Owned`. These
+are both presented as equal options, and this RFC does not propose either as a
+superior option. It is proposed that both be implemented, and the more popular
+choice be chosen during the stabilization period.
+
+Option A is as follows:
+
+`Cow::Owned` would be exported as U+1F42E COW FACE, `🐮`. The choice of this
+identifier should be clear -- A value of variant `🐮` has been copied for write,
+or "cowed". We don't believe that the similarities between 🐄 and 🐮 will be
+confusing in practice, all Rust developers will clearly pronounce the former
+"cow" and the latter "owned/cowed".
+
+`Cow::Borrowed` would be exported as U+262D HAMMER AND SICKLE, `☭`. This is
+because ownership is theft. No further explanation is required.
+
+The main drawback to the use of these characters is that 🐮 resides outside of
+the Unicode BMP, and thus requires multiple code units to represent in UTF-16.
+Because UTF-16 is a fixed width encoding, this may be confusing to many
+developers. For this reason, the following alternative is proposed:
+
+As no identifier or glyph today meets the needs of sufficiently representing the
+embodiment of `Borrowed` and `Owned`, while being sufficiently unique from
+identifiers which may be used in Rust today, two characters from the Unicode
+private use area will be used instead.
+
+`Borrowed` will be exported as U+FF8E, and `Owned` will be exported as U+FF8F.
```

Comment:
ﾎ and ﾏ do not look like private use characters. The PUA in BMP is U+E000 – U+**F8FF**.